src/status_im2/contexts/chat/bottom_sheet_composer/effects.cljs

```diff
@@ -0,0 +1,91 @@
+
+(defn reenter-screen-effect
+  [{:keys [text-value saved-cursor-position maximized?]}
+   {:keys [content-height]}
+   {:keys [input-content-height input-text input-maximized?]}]
+  (when (and (empty? @text-value) (not= input-text nil))
+    (reset! text-value input-text)
+    (reset! content-height input-content-height)
+    (reset! saved-cursor-position (count input-text)))
+  (when input-maximized?
+    (reset! maximized? true)))
+
+(defn maximized-effect
+  [{:keys [maximized?]}
+   {:keys [height saved-height last-height]}
+   {:keys [max-height]}
+   {:keys [input-content-height]}]
+  (when (or @maximized? (>= input-content-height max-height))
+    (reanimated/animate height max-height)
+    (reanimated/set-shared-value saved-height max-height)
+    (reanimated/set-shared-value last-height max-height)))
+
+(defn refocus-effect
+  [{:keys [input-ref]}
+   {:keys [input-refocus?]}]
+  (when input-refocus?
+    (.focus ^js @input-ref)
+    (rf/dispatch [:chat.ui/set-input-refocus false])))
+
+(defn layout-effect
+  [{:keys [lock-layout?]}]
+  (when-not @lock-layout?
+    (js/setTimeout #(reset! lock-layout? true) 500)))
+
+(defn kb-default-height-effect
+  [{:keys [kb-default-height]}]
+  (when-not @kb-default-height
+    (async-storage/get-item :kb-default-height
+                            #(reset! kb-default-height (when (not= nil %) (js/parseInt %))))))
+
+(defn background-effect
+  [{:keys [maximized?]}
+   {:keys [opacity background-y]}
+   {:keys [max-height]}
+   {:keys [input-content-height]}]
+  (when (or @maximized? (>= input-content-height (* max-height c/background-threshold)))
+    (reanimated/set-shared-value background-y 0)
+    (reanimated/animate opacity 1)))
+
+(defn images-effect
+  [{:keys [container-opacity]}
+   images?]
+  (when images?
+    (reanimated/animate container-opacity 1)))
+
+(defn empty-effect
+  [{:keys [text-value maximized? focused?]}
+   {:keys [container-opacity]}
+   images?]
+  (when (and (empty? @text-value) (not images?) (not @maximized?) (not @focused?))
+    (reanimated/animate-delay container-opacity 0.7 200)))
+
+(defn component-will-unmount
+  [{:keys [keyboard-show-listener keyboard-hide-listener keyboard-frame-listener]}]
+  (.remove ^js @keyboard-show-listener)
+  (.remove ^js @keyboard-hide-listener)
+  (.remove ^js @keyboard-frame-listener))
+
+(defn use-effect
+  [props state animations {:keys [max-height] :as dimensions} chat-input keyboard-height images?]
+  (rn/use-effect
+   (fn []
+     (maximized-effect state animations dimensions chat-input)
+     (refocus-effect props chat-input)
+     (reenter-screen-effect state dimensions chat-input)
+     (layout-effect state)
+     (kb-default-height-effect state)
+     (background-effect state animations dimensions chat-input)
+     (images-effect animations images?)
+     (empty-effect state animations images?)
+     (kb/add-kb-listeners props state animations dimensions keyboard-height)
+     #(component-will-unmount props))
```

Comment:
Love that you created a function for component-did-mount to make it's implementation in hooks more idiomatic, Personally I think this be added to our standards

Reply:
This is a `useEffect` hook which is different from component did mount. `useEffect` would be called on mount similar to `componentDidMount`, but will also be called whenever any of the values in the dependecy array changes (in this case `maxHeight`). The last function in `useEffect` is similar to `componentWillUnmount`